### DISCUSSION TITLE
FVM support

### DISCUSSION THREAD

**Discussion by rocboronat** [codemagic-ci-cd/codemagic-docs#1358] [Ideas]
Hi CodeMagic team :wave:

I made a question about FVM on Slack and one member of the Slack team redirected me to here, telling me I should mention Jukka (@Ycoi on Slack I guess?).

The request is about CodeMagic supporting FVM. It would be great if our development team doesn’t need to update anything in CodeMagic scripts when we change the Flutter version our project relies on. It would be great if CodeMagic just used the FVM configuration file to know which version of Flutter use.

In case you don't know, the FVM config file looks like this:
```json
{
  "flutterSdkVersion": "2.10.5",
  "flavors": {}
}
```
Is it possible for you to get the Flutter version in use from this FVM config file? It would be a.w.e.s.o.m.e.

Thank you so much :slightly_smiling_face:

**Comment by tsimbalar** [comment]
For reference, in case that is useful, in the GitHub Actions world, there is an action that parses the `fvm` config file to expose the desired version ... which you can then use to configure the "flutter install" action

see https://github.com/kuhnroyal/flutter-fvm-config-action

**Comment by Ycoi** [comment]
Thanks for creating this topic @rocboronat ! It helps us in tracking this feature request and also in seeing how much possible demand there would be for this. If someone reads this and would also like to see this happen, please upwote on the original post. 

In the meanwhile I'll discuss this internally with the team and see how difficult something like this would be to implement.

**Reply by rocboronat** [reply]
Thank you so much @Ycoi 😊

**Comment by Ycoi** [comment]
@rocboronat I wanted to let you know that this is now possible both for [codemagic.yaml](https://docs.codemagic.io/yaml-quick-start/building-a-flutter-app/#setting-the-flutter-version) and [WFE](https://docs.codemagic.io/flutter-configuration/flutter-projects/#setting-the-flutter-version) builds. Let us know if you run into any issues!

**Reply by rocboronat** [reply]
Lovely! I'll try next Tuesday ✌
